src/main/java/com/google/api/generator/gapic/model/Message.java

```diff
@@ -30,6 +30,9 @@
 public abstract class Message {
   public abstract String name();
 
+  // The fully-qualified proto name, which differs from the Java fully-qualified name.
+  public abstract String fullProtoName();
```

Comment:
Can you please specify how exactly they are different? Is it about the extra `com.` prefix for java names, or more than that?

Reply:
Yes, it's the proto package versus the `java_package`. Added an example.